src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -613,43 +613,55 @@ void Replayer<I>::scan_remote_mirror_snapshots(
     m_remote_snap_id_end = remote_snap_id;
     m_remote_mirror_snap_ns = *mirror_ns;
   }
-  image_locker.unlock();
-
-  unlink_snap_ids.erase(m_remote_snap_id_start);
-  unlink_snap_ids.erase(m_remote_snap_id_end);
-  if (!unlink_snap_ids.empty()) {
-    locker->unlock();
 
-    // retry the unlinking process for a remote snapshot that we do not
-    // need anymore
-    auto remote_snap_id = *unlink_snap_ids.begin();
-    dout(10) << "unlinking from remote snapshot " << remote_snap_id << dendl;
-    unlink_peer(remote_snap_id);
-    return;
+  if (m_remote_snap_id_start != 0 &&
+      remote_image_ctx->snap_info.find(m_remote_snap_id_start) ==
+        remote_image_ctx->snap_info.end()) {
```

Comment:
can be ignored: `remote_image_ctx->snap_info.count(m_remote_snap_id_start) == 0` may be more reedable?

Reply:
Sure, why not.